File: 프로그래머스/Lv.0/캐릭터의 좌표/120861.js
function solution(keyinput, board) {
    const answer = [0, 0];
    
    keyinput.forEach(v => {
       if (v === "up") {
           if (answer[1] >= (board[1]-1) / 2) return;
           answer[1]++;
       } else if (v === "down") {
           if (answer[1] <= (board[1]-1) / 2 * -1) return;
           answer[1]--;
       } else if (v === "left") {
           if (answer[0] <= (board[0]-1) / 2 * -1) return;
           answer[0]--;
       } else if (v === "right") {
           if (answer[0] >= (board[0]-1) / 2) return;
           answer[0]++;
       }
    });
    
    return answer;
}

File: 프로그래머스/Lv.0/캐릭터의 좌표/120861.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        
        for (String key : keyinput) {
            if ("up".equals(key)) {
                if (answer[1] >= board[1] / 2) continue;
                answer[1]++;
            } else if ("down".equals(key)) {
                if (answer[1] <= board[1] / 2 * -1) continue;
                answer[1]--;
            } else if ("left".equals(key)) {
                if (answer[0] <= board[0] / 2 * -1) continue;
                answer[0]--;
            } else if ("right".equals(key)) {
                if (answer[0] >= board[0] / 2) continue;
                answer[0]++;
            }
        }
        
        return answer;
    }
}
